near-rust-allocator-proxy/src/allocator.rs

```diff
@@ -1,26 +1,26 @@
 use backtrace::Backtrace;
+use once_cell::sync::Lazy;
 use std::alloc::{GlobalAlloc, Layout};
 use std::cell::Cell;
 use std::fs::OpenOptions;
 use std::io::Write;
+use std::mem;
 use std::os::raw::c_void;
 use std::ptr::null_mut;
 use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
-use std::{fs, mem};
+use std::sync::Mutex;
 
 const MEBIBYTE: usize = 1024 * 1024;
-const REPORT_USAGE_INTERVAL: usize = 512 * MEBIBYTE;
 const SKIP_ADDR: *mut c_void = 0x700000000000 as *mut c_void;
+/// Configure how often should we print stack trace, whenever new record is reached.
+pub(crate) static REPORT_USAGE_INTERVAL: AtomicUsize = AtomicUsize::new(512 * MEBIBYTE);
 /// Should be a configurable option.
-static SAVE_STACK_TRACES_TO_FILE: AtomicBool = AtomicBool::new(false);
+pub(crate) static SAVE_STACK_TRACES_TO_FILE: AtomicBool = AtomicBool::new(false);
 /// Should be a configurable option.
-#[cfg(target_os = "linux")]
-static ENABLE_STACK_TRACE: AtomicBool = AtomicBool::new(true);
-const LOGS_PATH: &str = "/tmp/logs";
-
-// Currently there is no point in getting stack traces on non-linux platform, because other tools don't support linux.
-#[cfg(not(target_os = "linux"))]
-const ENABLE_STACK_TRACE: AtomicBool = AtomicBool::new(false);
+pub(crate) static ENABLE_STACK_TRACE: AtomicBool = AtomicBool::new(false);
+/// TODO: Make this configurable.
+pub(crate) static LOGS_PATH: Lazy<Mutex<String>> =
```

Comment:
This shold be `PathBuf` rather than `String`

Reply:
And you probably want to use `OnceCell` directly here, rather than Lazy, to be able to configure it.

Reply:
Actually, that code is not longer needed. The new `rust-memory-analyzer` crate will be able to get symbols without need for process to save stack traces to file.

https://github.com/near/near-memory-tracker/compare/piotr-remove-saving-traces-to-file